### PR TITLE
Migrate to another package of apache http client library

### PR DESCRIPTION
**Pull request by rpdome**
Related PR: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1415

### CODE REVIEW

File: msal/build.gradle [hunk 1]
```diff
@@ -280,7 +281,7 @@ dependencies {
     testImplementation "org.mockito:mockito-inline:$rootProject.ext.mockitoCoreVersion"
     testImplementation ("org.robolectric:robolectric:$rootProject.ext.robolectricVersion")
     // For mockito!
-    testImplementation group: 'org.apache.httpcomponents', name: 'httpclient', version: '4.5.13'
+    testImplementation "cz.msebera.android:httpclient:$rootProject.ext.msebraApacheHttpClientVersion"
```
Comment (shahzaibj) [inline msal/build.gradle:284]: implementation should carry over to test dependencies as well...why do we need this explicitly?

File: gradle/versions.gradle [hunk 2]
```diff
@@ -41,7 +41,7 @@ ext {
     rulesVersion = "1.2.0"
     robolectricVersion = "4.3.1"
     uiAutomatorVersion = "2.2.0"
-    apacheHttpVersion = "4.5.13"
+    msebraApacheHttpClientVersion = "4.5.8"
```
Comment (shahzaibj) [inline gradle/versions.gradle:44]:
```suggestion
    mseberaApacheHttpClientVersion = "4.5.8"
```

**Review by shahzaibj — APPROVED** [review]
left minor comment

**Review by AdamBJohnsonx — APPROVED** [review]

**Comment by rpdome** [comment]
FYI, @AdamBJohnsonx  @shahzaibj  I messed up the merging order in common. Had to resolve minor merge conflict due to common#1396.

**Pull request merged**